include/mockgyver.hrl

```diff
@@ -5,22 +5,53 @@
 
 -compile({parse_transform, mockgyver_xform}).
 
+-ifdef(OTP_RELEASE).
+-define(mockgyver_get_stacktrace__(),
+        try error(get_stack)
+        catch error:get_stack:St -> St
+        end).
+-else. % OTP_RELEASE
+-define(mockgyver_get_stacktrace__(),
+        try error(get_stack)
+        catch error:get_stack -> erlang:get_stacktrace()
+        end).
+-endif. % OTP_RELEASE.
+
+
+
 %% run tests with a mock
+-define(WITH_MOCKED_SETUP(SetupFun, CleanupFun, ForAllTimeout, PerTcTimeout,
+                          Tests, MockOpts),
+        %% Wrap it in a fun so we can have a few local variables
+        fun() ->
+                StackId = erlang:phash2(?mockgyver_get_stacktrace__()),
+                Base = #{num_sessions => length(Tests),
+                         signature => {?FILE, ?LINE, ?FUNCTION_NAME, StackId}},
+                IndexedTests = lists:zip(lists:seq(1, length(Tests)),
+                                         Tests),
+                {timeout, ForAllTimeout,           %% timeout for all tests
+                 [{timeout, PerTcTimeout,          %% timeout for each test
+                   {spawn,
+                    {atom_to_list(__Test),         %% label per test
+                     fun() ->
```

Comment:
This no longer uses `?WITH_FUN` from eunit_addons due to the need for indexing. Is it desirable to pursue somehow extending eunit_addons to cover this usecase? if yes, then how?

Reply:
There's no self-interest in using eunit_addons as such, but I'll have a deeper look at the changes.